dataactbroker/handlers/dashboard_handler.py

```diff
@@ -19,6 +23,63 @@
 FILE_TYPES = ['A', 'B', 'C', 'cross-AB', 'cross-BC', 'cross-CD1', 'cross-CD2']
 
 
+def list_rule_labels(files, error_level='warning', fabs=False):
+    """ Returns a list of rule labels based on the files and error type provided
+
+        Args:
+            files: A list of files for which to return rule labels. If blank, return all matching other arguments
+            error_level: A string indicating whether to return errors, warnings, or both. Defaults to warning
+            fabs: A boolean indicating whether to return FABS or DABS rules. Defaults to False
+
+        Returns:
+            JsonResponse of the rule labels the arguments indicate. JsonResponse error if invalid file types are
+            provided or any file types are provided for FABS
+    """
+    # Make sure list is empty when requesting FABS rules
+    if fabs and len(files) > 0:
+        return JsonResponse.error(ValueError('Files list must be empty for FABS rules'), StatusCode.CLIENT_ERROR)
+
+    invalid_files = [invalid_file for invalid_file in files if invalid_file not in FILE_TYPES]
+    if invalid_files:
+        return JsonResponse.error(ValueError('The following are not valid file types: {}'.
+                                             format(', '.join(invalid_files))),
+                                  StatusCode.CLIENT_ERROR)
```

Comment:
Are there any guidelines on when to use `ResponseException` vs `JsonResponse.error`? I see both used interchangeably in the code.

Reply:
I try to use `JsonResponse.error` when it's some sort of input error and `ResponseException` if there's an actual problem if possible, but that's a personal preference and we don't have any real guidelines for it